service/src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -271,6 +283,7 @@ public ResponseEntity<Void> grantRole(
       throw new InvalidRoleException(
           "Users cannot grant role APPLICATION. Use application registration instead.");
     }
+    // No additional authz check as this is just a wrapper around a Sam endpoint.
```

Comment:
For consistency, I think it would make sense to move auth check from SamService to here...

Same for other SamService calls

Reply:
ah, I missed the `checkAuthz` calls in SamService, good catch! 

The intended pattern here is that explicit auth checks are in the controller, but any call to Sam (like adding/removing a user) implicitly also checks authz.
Those authz checks were always redundant calls - there's no need to explicitly ask Sam if user A can add user B to the workspace, since the next call is "add user B to the workspace using user A's credentials", and `addUserToPolicy` will just throw an exception if the caller doesn't have permission.

Reply:
I'm actually fine with having controllers call SamService directly, it seems reasonable that controllers might need to call multiple different services.
The calls going directly from controller -> SamService are generally ones where WSM is really just wrapping Sam endpoints, so internally wrapping those in another service seems like more hassle than it's worth to me